### PR TITLE
Detect node test runner

### PR DESCRIPTION
**Pull request by fabnguess**

### CODE REVIEW

File: src/utils/getTestFrameworkName.ts [hunk 1]
```diff
@@ -17,3 +24,19 @@ export function getTestFrameworkName(deps = {}) {
 
   return "N/A";
 }
+
+function IsNodeTestingUsed() {
```
Comment (PierreDemailly) [inline src/utils/getTestFrameworkName.ts]: Functions should be camelCase

```suggestion
function isNodeTestingUsed() {
```
or
```suggestion
function isNodeTestRunnerUsed() {
```

File: src/utils/getTestFrameworkName.ts [hunk 2]
```diff
@@ -17,3 +24,19 @@ export function getTestFrameworkName(deps = {}) {
...
+  const [majorVersion] = process.versions.node.split(".").map(Number);
+  if (majorVersion <= 18) {
+    return false;
+  }
+
+  const nodeTestRegex = /(?:node\s+--test|tsx)/;
```
Comment (PierreDemailly) [inline src/utils/getTestFrameworkName.ts]: Use file const prefixed with `k` (`kNodeTestRunnerRegex`)

File: src/utils/getTestFrameworkName.ts [hunk 3]
```diff
@@ -1,4 +1,11 @@
+// Import Node.js Dependencies
+import fs from "fs";
+import path from "path";
```
Comment (PierreDemailly) [inline src/utils/getTestFrameworkName.ts]:
```suggestion
import fs from "node:fs";
import path from "node:path";
```
(This should even be removed as we will not parse package.json again)

**Review by PierreDemailly — COMMENTED** [review]
Could you add tests for `getTestFrameworkName` ?

File: src/utils/getTestFrameworkName.ts [hunk 4]
```diff
@@ -17,3 +24,19 @@ export function getTestFrameworkName(deps = {}) {
...
+
+function IsNodeTestingUsed() {
+  const [majorVersion] = process.versions.node.split(".").map(Number);
+  if (majorVersion <= 18) {
+    return false;
+  }
+
+  const nodeTestRegex = /(?:node\s+--test|tsx)/;
+
+  const packageJsonPath = path.join(process.cwd(), "package.json");
```
Comment (PierreDemailly) [inline src/utils/getTestFrameworkName.ts]: We should not parse package.json again, scripts should be accessible from here:
https://github.com/dashlog/core/blob/6159d526453b209542bbc7df434d3637f4c34f1c/src/services/repository.ts#L155-L164

File: src/utils/getTestFrameworkName.ts [hunk 5]
```diff
@@ -17,3 +28,13 @@ export function getTestFrameworkName(deps = {}) {
 
   return "N/A";
 }
+
+function isNodeTestRunnerUsed(scripts : Record<string, string>) : boolean {
+  const [majorVersion] = process.versions.node.split(".").map(Number);
+  if (majorVersion <= 18) {
+    return false;
+  }
```
Comment (fraxken) [inline src/utils/getTestFrameworkName.ts]: We don't need that, here you'r checking the Node.js version currently running by the dashlog process (not the Node.js version supported by the project we are scanning)

**Review by fraxken — CHANGES_REQUESTED** [review]

**Review by fraxken — APPROVED** [review]

File: src/utils/getTestFrameworkName.ts [hunk 6]
```diff
@@ -1,4 +1,15 @@
-export function getTestFrameworkName(deps = {}) {
+// Import Third-party Dependencies
+import type { PackageJSON } from "@nodesecure/npm-types";
+
+// CONSTANTS
+const kNodeTestRunnerRegex = /(?:node\s+--test|tsx)/;
+
+export function getTestFrameworkName(packageJson : PackageJSON) : string {
```
Comment (PierreDemailly) [inline src/utils/getTestFrameworkName.ts]:
```suggestion
export function getTestFrameworkName(packageJson: PackageJSON): string {
```

**Review by PierreDemailly — APPROVED** [review]

**Review by fraxken — APPROVED** [review]

**Comment by fraxken** [comment]
@allcontributors please add @fabnguess for code

**Pull request merged**